### PR TITLE
Ravinder | Test | Integrating angular material in ledger module

### PR DESCRIPTION
**Pull request by ravinderwalkover**
https://app.clickup.com/t/4dfty7

### CODE REVIEW

File: apps/web-giddh/src/app/material-ledger/components/share-ledger/share-ledger.component.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+.btn-success:disabled {
+    color: $white !important;
+}
+.mat-form-field {
+    width: 80% !important;
+    margin-top: 2px;
+}
+.pd-0 {
+    padding-left: 5px !important;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/material-ledger/components/share-ledger/share-ledger.component.scss:11]: This is not as per convention as it is for `padding: 0`. Consider renaming the class for padding.

File: apps/web-giddh/src/app/material-ledger/ledger.vm.ts [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+}
+
+/** List of available ITC */
+export const AVAILABLE_ITC_LIST = [
+    { label: 'Import of goods', value: AvailableItc.ImportOfGoods },
+    { label: 'Import of services', value: AvailableItc.ImportOfServices },
+    { label: 'Others', value: AvailableItc.Others }
+];
+
+export const MaterialColorPalette = "primary";
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/material-ledger/ledger.vm.ts:490]: Do use SNAKE_CASE for constants

File: apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    styleUrls: ['./confirm-modal.component.scss']
+})
+
+export class ConfirmModalComponent {
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public inputData,
+        public dialogRef: MatDialogRef<any>) {
+    }
+
+    public sendResponse(response: boolean) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts:16]: Do add JS doc and add return type

File: apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+    selector: 'new-confirm-modal',
+    templateUrl: './confirm-modal.component.html',
+    styleUrls: ['./confirm-modal.component.scss']
+})
+
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts:9]: Avoid empty line between `@Component` decorator and class declaration

File: apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+    selector: 'new-confirm-modal',
+    templateUrl: './confirm-modal.component.html',
+    styleUrls: ['./confirm-modal.component.scss']
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts:7]: Do add `ChangeDetectionStrategy.OnPush` for new components

File: apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    styleUrls: ['./confirmation-modal.component.scss']
+})
+
+export class NewConfirmationModalComponent {
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public inputData,
+        public dialogRef: MatDialogRef<any>) {
+    }
+
+    public sendResponse(response: any) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts:16]: Do add JS doc and return type

File: apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+    selector: 'new-confirmation-modal',
+    templateUrl: './confirmation-modal.component.html',
+    styleUrls: ['./confirmation-modal.component.scss']
+})
+
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts:9]: Avoid empty line between `@Component` decorator and class declaration

File: apps/web-giddh/src/assets/css/style-1.scss [hunk 8]
```diff
@@ -2749,3 +2749,68 @@ form#groupDetailForm {
...
+}
+.ledger-advance-search-field {
+    .mat-form-field-suffix {
+        right: 10px !important;
+    }
+    input.mat-input-element {
+        padding-left: 8px !important;
+        width: 96%;
+    }
+    input.mat-input-element.date-field, input.mat-input-element.description-field {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/assets/css/style-1.scss:2810]: Consider nesting this to selector `input.mat-input-element` at line 2806
```suggestion
    &.date-field, &.description-field {
```

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 9]
```diff
@@ -1165,12 +1172,12 @@ modal-container.modal {
     color: $nobel;
 }
 .price-input {
-    border: 1px solid $gainsboro;
-    font-size: 14px;
-    background: $white;
-    padding: 1px 2px;
+    border: 1px solid $gainsboro !important;
+    font-size: 14px !important;
+    background: #ffffff !important;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/assets/css/style-2.scss:1177]: Please check color variable change is getting removed.
```suggestion
    background: $white!important;
```

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 10]
```diff
@@ -8019,10 +8027,10 @@ input.mat-input-element {
 .mat-form-field-underline {
     display: none !important;
 }
-.cdk-overlay-container,
-.cdk-global-overlay-wrapper {
-    z-index: 99999;
-}
+// .cdk-overlay-container,
```
Comment (prateek-giddh) [inline apps/web-giddh/src/assets/css/style-2.scss:8030]: Consider removing the commented code

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 11]
```diff
@@ -10173,3 +10181,13 @@ mat-year-view table tr td.mat-calendar-body-label {
     display: none !important;
 }
 /** Angular material datepicker **/
+.pd-0 {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/assets/css/style-2.scss:10184]: We already have `p-0` class for setting 0 padding

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/invoice/templates/edit-template/modals/confirmation-modal/confirmation.modal.module.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import { NgModule } from "@angular/core";
+import { DeleteTemplateConfirmationModelComponent } from "./confirmation.modal.component";
+
+@NgModule({
+    declarations: [
+        DeleteTemplateConfirmationModelComponent
+    ],
+    imports: [
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/invoice/templates/edit-template/modals/confirmation-modal/confirmation.modal.module.ts:8]: Done

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 13]
```diff
@@ -0,0 +1,322 @@
...
+                                <small>{{localeData?.default_vouchers_selected}}</small>
+                            </p>
+                        </div>
+                        <div class="col-md-7">
+                            <div class="row">
+                                <div class="pr-0 col-sm-3 col-12 switch-wrapper">
+                                    <div class="checkbox square-switch">
+                                        <mat-slide-toggle [color]="materialColorPalette" id="includeVouchers" formControlName="includeVouchers">
+                                            <span class="pull-left"
+                                                *ngIf="advanceSearchForm.get( 'includeVouchers').value">{{commonLocaleData?.app_include}}</span>
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:121]: Done

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 14]
```diff
@@ -0,0 +1,322 @@
...
+                                            <span class="pull-right"
+                                                *ngIf="!advanceSearchForm.get( 'includeVouchers').value">{{commonLocaleData?.app_exclude}}</span>
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:123]: Done

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 15]
```diff
@@ -0,0 +1,322 @@
...
+                    <div class="form-group toggle-btn mr-b05">
+                        <label class="cp" (click)="toggleOtherDetails()">
+                            <i class="cp collapse-icons" aria-hidden="true"
+                                [ngClass]="{ 'icon-minus': showOtherDetails, 'icon-plus': !showOtherDetails}"></i>{{commonLocaleData?.app_other_details}}
+                        </label>
+                    </div>
+                </div>
+
+                <!-- other details container -->
+                <ng-container *ngIf="advanceSearchForm.get( 'includeDescription').value">
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:192]: Done

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 16]
```diff
@@ -0,0 +1,322 @@
...
+                                <div class="col-sm-5">
+                                    <label>{{commonLocaleData?.app_inventory}}</label>
+                                </div>
+                                <div class="col-sm-7" formGroupName="inventory">
+                                    <div class="row">
+                                        <div class="pr-0 col-sm-3 col-12 switch-wrapper">
+                                            <div class="checkbox square-switch">
+                                                <mat-slide-toggle [color]="materialColorPalette" id="includeInventory" formControlName="includeInventory">
+                                                    <span class="pull-left"
+                                                        *ngIf="advanceSearchForm.get( 'inventory.includeInventory').value">{{commonLocaleData?.app_include}}</span>
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:205]: Done

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 17]
```diff
@@ -0,0 +1,322 @@
...
+                                                    <span class="pull-right"
+                                                        *ngIf="!advanceSearchForm.get( 'inventory.includeInventory').value">{{commonLocaleData?.app_exclude}}</span>
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:207]: Done

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 18]
```diff
@@ -0,0 +1,322 @@
...
+                                                </mat-slide-toggle>
+                                            </div>
+                                        </div>
+                                        <div class="col-sm-9 col-12">
+                                            <div class="form-group">
+                                                <sh-select [options]="stockListDropDown$ | async" name="inventory"
+                                                    (onClear)="onDDClear( 'inventory')" formControlName="inventories"
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:214]: Done

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html [hunk 19]
```diff
@@ -0,0 +1,322 @@
...
+                                                    [placeholder]="commonLocaleData?.app_quantity">
+                                                </mat-form-field>    
+                                            </div>
+                                        </div>
+                                    </div>
+                                    <div class="row">
+                                        <div class="col-6 col-md-6">
+                                            <div class="form-group">
+                                                <sh-select [showClear]="false" [width]="'100%'" formControlName="inventoryVal"
+                                                    (selected)="onRangeSelect( 'inventoryVal', $event)"
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.html:246]: Done

File: apps/web-giddh/src/app/material-ledger/components/generate-voucher-confirm-modal/generate-voucher-confirm-modal.component.ts [hunk 20]
```diff
@@ -0,0 +1,19 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+    selector: 'generate-voucher-confirmation-modal',
+    templateUrl: './generate-voucher-confirm-modal.component.html',
+    styleUrls: ['./generate-voucher-confirm-modal.component.scss']
+})
+
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/generate-voucher-confirm-modal/generate-voucher-confirm-modal.component.ts:9]: Done

File: apps/web-giddh/src/app/material-ledger/components/generate-voucher-confirm-modal/generate-voucher-confirm-modal.component.ts [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+export class GenerateVoucherConfirmationModalComponent {
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public inputData,
+        public dialogRef: MatDialogRef<any>) {
+    }
+
+    public sendResponse(response: boolean) {
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/generate-voucher-confirm-modal/generate-voucher-confirm-modal.component.ts:16]: Done

File: apps/web-giddh/src/app/material-ledger/components/share-ledger/share-ledger.component.scss (hunk 1 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/share-ledger/share-ledger.component.scss:11]: We created a new class pd-0 in style-2.scss for padding: 0 and we are overriding it here.

File: apps/web-giddh/src/app/material-ledger/ledger.vm.ts (hunk 2 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/ledger.vm.ts:490]: Done

File: apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts (hunk 3 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts:16]: Done

File: apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts (hunk 4 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts:9]: Done

File: apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts (hunk 6 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts:16]: Done

File: apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts (hunk 7 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/theme/new-confirmation-modal/confirmation-modal.component.ts:9]: Done

File: apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts (hunk 5 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/theme/new-confirm-modal/confirm-modal.component.ts:7]: Done

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 9 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/assets/css/style-2.scss:1177]: Done

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 10 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/assets/css/style-2.scss:8030]: Done

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 11 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/assets/css/style-2.scss:10184]: that class is getting overridden if we are using form-control class

File: apps/web-giddh/src/app/services/toaster.service.ts [hunk 22]
```diff
@@ -44,4 +46,14 @@ export class ToasterService {
     public clearAllToaster(): void {
         this._toaster.clear();
     }
+
+    public showSnackBar(type: string, message: string, title: string = APP_DEFAULT_TITLE): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/services/toaster.service.ts:50]: Do add JS doc

File: apps/web-giddh/src/app/theme/snackbar/snackbar.module.ts [hunk 23]
```diff
@@ -0,0 +1,18 @@
+import { NgModule } from "@angular/core";
+import { SnackBarComponent } from "./snackbar.component";
+
+@NgModule({
+    declarations: [
+        SnackBarComponent
+    ],
+    imports: [
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/snackbar/snackbar.module.ts:8]: Avoid empty `imports` statement

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

**Review by prateek-giddh — APPROVED** [review]

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.scss [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+        }
+    }
+}
+@media screen and (max-width: 575px) {
+    .switch-wrapper {
+        padding-right: 15px !important;
+        margin-bottom: 15px;
+    }
+}
+.mat-switch {
```
Comment (aditi-tiwari17) [inline apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.scss:92]: please add this css above media queries and please do not add any css after media queries in any page.

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.scss [hunk 25]
```diff
@@ -0,0 +1,28 @@
...
+        flex-direction: column;
+        span {
+            margin-right: 15px !important;
+        }
+    }
+    .show-columnar-button {
+        display: none;
+    }
+}
+textarea.receipient-email {
```
Comment (aditi-tiwari17) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.scss:25]: please add this css above media queries

File: apps/web-giddh/src/app/material-ledger/components/ledger-aside-pane/component/ledger-aside-pane-account/ledger-aside.pane.account.component.scss [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+@media (max-width: 320px) {
+    :host {
+        max-width: 280px;
+    }
+    :host .aside-pane {
+        max-width: 280px;
+    }
+}
+
+.flexy-child {
```
Comment (aditi-tiwari17) [inline apps/web-giddh/src/app/material-ledger/components/ledger-aside-pane/component/ledger-aside-pane-account/ledger-aside.pane.account.component.scss:85]: please add this css above media queries

File: apps/web-giddh/src/app/material-ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.scss [hunk 27]
```diff
@@ -0,0 +1,232 @@
...
+        -webkit-transform: rotate(360deg);
+    }
+}
+@keyframes spin {
+    100% {
+        -webkit-transform: rotate(360deg);
+        transform: rotate(360deg);
+    }
+}
+#new-ledger-enter-amount,
```
Comment (aditi-tiwari17) [inline apps/web-giddh/src/app/material-ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.scss:206]: please add this css above keyframes

File: apps/web-giddh/src/app/material-ledger/ledger.component.scss [hunk 28]
```diff
@@ -0,0 +1,1216 @@
...
+}
+
+@media (min-width: 768px) and (max-width: 991px) {
+    .ledger-datepicker {
+        img {
+            padding-top: 5px;
+        }
+    }
+}
+.icon-upload {
```
Comment (aditi-tiwari17) [inline apps/web-giddh/src/app/material-ledger/ledger.component.scss:1175]: please add this css above media queries

File: apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.scss (hunk 24 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/advance-search/advance-search.component.scss:92]: Done

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.scss (hunk 25 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.scss:25]: Done

File: apps/web-giddh/src/app/material-ledger/components/ledger-aside-pane/component/ledger-aside-pane-account/ledger-aside.pane.account.component.scss (hunk 26 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/ledger-aside-pane/component/ledger-aside-pane-account/ledger-aside.pane.account.component.scss:85]: This is existing css, i didn't add it. Still i moved it above

File: apps/web-giddh/src/app/material-ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.scss (hunk 27 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/material-ledger/components/new-ledger-entry-panel/new-ledger-entry-panel.component.scss:206]: Done

**Pull request merged**